cvat/apps/dataset_manager/bindings.py

```diff
@@ -567,14 +563,27 @@ def _import_track(self, track, parent_label_id=None):
                 for attrib in shape['attributes']
                 if self._get_mutable_attribute_id(label_id, attrib.name)
             ]
-        # TODO: remove once importers are guaranteed to return correct type
-        # (see https://github.com/cvat-ai/cvat/pull/8226/files#r1695445137)
-            points = shape["points"]
-            for i, point in enumerate(map(float, points)):
-                points[i] = point
+            self._ensure_points_converted_to_floats(shape)
 
         return _track
 
+    def _ensure_points_converted_to_floats(self, shape) -> None:
+        """
+        Historically, there were importers that were not converting points to ints/floats.
+        The only place to make sure that all points in shapes have the right type was this one.
+        However, this does eat up a lot of memory for some reason.
+        (see https://github.com/cvat-ai/cvat/pull/8226/files#r1695445137)
+
+        So, before we can guarantee that all the importers are returning the right data,
+        we have to have this conversion.
+        """
+        # if points is LazyList or tuple, we can be sure it has the right type already
+        if isinstance(points := shape["points"], LazyList | tuple):
```

Comment:
How a tuple guarantees the correct type? Wouldn't be better to go through the input and check types instead?